activemq-client/src/main/java/org/apache/activemq/transport/discovery/masterslave/MasterSlaveDiscoveryAgent.java

```diff
@@ -24,7 +24,10 @@
 /**
  * A static DiscoveryAgent that supports connecting to a Master / Slave tuple
  * of brokers.
+ *
+ * @deprecated This class is superseded by HADiscoveryAgent and will be removed in 5.18.0.
  */
+@Deprecated
```

Comment:
Can it use the 'for removal' flag? If so it seems like it should. Though i realise most things arent going to touch the code itself so it would really only be for descriptive purposes.

(I've yet to use it myself so not clear when it came in)